File: Demo/Sources/ViewControllers/TextField/TextFieldViewController.m
//
//  Globus
//

#import "TextFieldViewController.h"

@interface TextFieldViewController ()

@property(nonatomic, weak) IBOutlet GLBTextField* textField;

@end

@implementation TextFieldViewController

#pragma mark - UIViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    GLBTextStyle* textStyle = [GLBTextStyle new];
    textStyle.font = [UIFont boldSystemFontOfSize:16.0f];
    textStyle.color = UIColor.darkGrayColor;
    _textField.textStyle = textStyle;
    
    GLBTextStyle* placeholderStyle = [GLBTextStyle new];
    placeholderStyle.font = [UIFont boldSystemFontOfSize:16.0f];
    placeholderStyle.color = UIColor.lightGrayColor;
    _textField.placeholderStyle = placeholderStyle;
    
    _textField.placeholder = @"Placeholder";
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
    
    [_textField becomeFirstResponder];
}

#pragma mark - Actions

- (IBAction)pressedMenu:(id)sender {
    [self.glb_slideViewController showLeftViewControllerAnimated:YES complete:nil];
}

- (IBAction)changedText:(id)sender {
    NSLog(@"ChangedText: %@", _textField.text);
}

@end

File: Demo/Sources/ViewControllers/TextField/TextFieldViewController.swift
//
//  Globus
//

import Globus

class TextFieldViewController: GLBViewController {
    
    // MARK - Outlet property
    
    @IBOutlet weak var textField: GLBTextField!
    
    // MARK - UIViewController
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        let textStyle = GLBTextStyle.init()
        textStyle.font = UIFont.boldSystemFont(ofSize: 16)
        textStyle.color = UIColor.darkGray
        self.textField.textStyle = textStyle
        
        let placeholderStyle = GLBTextStyle.init()
        placeholderStyle.font = UIFont.boldSystemFont(ofSize: 16)
        placeholderStyle.color = UIColor.lightGray
        self.textField.placeholderStyle = placeholderStyle
        
        self.textField.placeholder = "Placeholder"
    }
    
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        
        self.textField.becomeFirstResponder()
    }
    
    // MARK - Action
    
    @IBAction internal func pressedMenu(_ sender: Any) {
        self.glb_slideViewController?.showLeftViewController(animated: true, complete: nil)
    }
    
    @IBAction internal func changedText(_ sender: Any) {
        print("ChangedText: \(self.textField.text)")
    }
    
    // MARK: - GLBNibExtension
    
    public override static func nibName() -> String {
        return "TextFieldViewController-Swift"
    }
    
}
